sklearn/manifold/tests/test_spectral_embedding.py

```diff
@@ -117,33 +134,59 @@ def test_spectral_embedding_two_components(seed=36):
     true_label[0:n_sample] = 1
 
     se_precomp = SpectralEmbedding(
-        n_components=1, affinity="precomputed", random_state=np.random.RandomState(seed)
+        n_components=1,
+        affinity="precomputed",
+        random_state=np.random.RandomState(seed),
+        eigen_solver=eigen_solver,
     )
-    embedded_coordinate = se_precomp.fit_transform(affinity)
-    # Some numpy versions are touchy with types
-    embedded_coordinate = se_precomp.fit_transform(affinity.astype(np.float32))
-    # thresholding on the first components using 0.
-    label_ = np.array(embedded_coordinate.ravel() < 0, dtype="float")
-    assert normalized_mutual_info_score(true_label, label_) == pytest.approx(1.0)
+    for dtype in [np.float32, np.float64]:
+        if eigen_solver == "amg" and not amg_loaded:
+            with pytest.raises(ValueError) as e:
+                embedded_coordinate = se_precomp.fit_transform(affinity.astype(dtype))
+            assert "The eigen_solver was set to 'amg', but" in str(e.value)
+        else:
+            embedded_coordinate = se_precomp.fit_transform(affinity.astype(dtype))
+            # thresholding on the first components using 0.
+            label_ = np.array(embedded_coordinate.ravel() < 0, dtype=np.float64)
+            assert normalized_mutual_info_score(true_label, label_) == pytest.approx(
+                1.0
+            )
```

Comment:
When passing `np.float32`, we should ensure that the fitted attributes (`embedding_` and `affinity_matrix_`) are resulting in a matrix with the same `dtype`. I assume that we want the same behaviour for the data returned by `fit_transform`. So we need to assert the `dtype` depending on the input `dtype`.

Reply:
the labels are integers, so on 32-bit OS they will be int32 no matter what the dtype is for the data

Reply:
The issue here is more about the old code using `dtype="float"` that will not be the same dtype depending on the platform. `np.int32` or `np.int64` would also be fine.

Reply:
@glemaitre could you please propose a specific commit to avoid any misunderstanding?